packages/net-stubbing/lib/server/route-matching.ts

```diff
@@ -121,3 +121,37 @@ export function getRouteForRequest (routes: BackendRoute[], req: CypressIncoming
     return _doesRouteMatch(route.routeMatcher, req)
   })
 }
+
+/**
+ * Is this a CORS preflight request that could be for an existing route?
+ * If there is a matching route with method = 'OPTIONS', returns false.
+ */
+export function matchesRoutePreflight (routes: BackendRoute[], req: CypressIncomingRequest) {
+  if (req.method !== 'OPTIONS' || !req.headers['access-control-request-method']) {
```

Comment:
I would factor this out into a little function, then your would avoid a need for the comment below

```js
const isPreflight = (req) => ...

export function matchesRoutePreflight (routes: BackendRoute[], req: CypressIncomingRequest) {
  if (!isPreflight(req)) {
    return false
  }
}
```